engine/src/exec-interface-stack.cpp

```diff
@@ -2182,3 +2183,29 @@ void MCStack::SetScriptOnly(MCExecContext& ctxt, bool p_script_only)
 {
     m_is_script_only = p_script_only;
 }
+
+// MERG-2015-10-11: [[ DocumentFilename ]] Add stack documentFilename property
+void MCStack::GetDocumentFilename(MCExecContext &ctxt, MCStringRef& r_document_filename)
+{
+    if (m_document_filename == nil)
```

Comment:
Initialize m_document_filename in the MCStack constructor to be MCValueRetain(kMCEmptyString) and this then simplifies this method.

If you just return, then 'r_document_filename' will be uninitialized and things will go wonky!